cmd/mtnormalise.cpp

```diff
@@ -52,9 +52,11 @@ void usage ()
    + "Example usage: mtnormalise wmfod.mif wmfod_norm.mif gm.mif gm_norm.mif csf.mif csf_norm.mif -mask mask.mif.";
 
   ARGUMENTS
-    + Argument ("input output", "list of all input and output tissue compartment files. See example usage in the description.").type_image_in().allow_multiple();
+    + Argument ("input output", "list of all input and output tissue compartment files. See example usage in the description.").type_various().allow_multiple();
 
   OPTIONS
+    + OptionGroup ("Options that affect the operation of the mtnormalise command")
```

Comment:
I'd rather the first section was left to default unless there's a very strong argument against. Here, these are the 'regular' options that users will likely want to use, the next section is indeed separate. But I reckon the standard section header is fine here.

Reply:
I tend to try to organise *all* groups into options as soon as there are option groups defined within the `cmd/` file. Most likely cause is that if you either locally define or import from a header file an option group, and then subsequently define a "root-level" option, that option will *appear* in the docs / help page to belong to the option group - or at least it has in the past, I don't know whether this has changed. I'll look into this more directly instead.

Reply:
In general, I agree with @jdtournier on this point; the very usage of the "default" section communicates an important point to the user, I reckon (as Donald essentially says).  However, in this very case, if we're reorganising the option categories for this command any way, I'd argue a slightly different organisation: the `-mask` and `-order` options as the only ones in the default option section, and the other ones (yet still the *non debugging ones*, those should be in the last debugging section, that is) in a section called something like "advanced options" (happily suggest a better name, this is just the general idea of it).  The reason is two-fold: first, `-mask` is actually mandatory for this command since the complete overhaul of the algorithm that Rami and I spent quite some time/effort on (the one that quite fundamentally changed it by taking it to the log domain, but also changed the nested iteration structure, added outlier rejection, etc...).  Second, the other options, in the case of this algorithm, you'd "*almost*" never touch (`-niter`, and `-value` or `-reference` as suggested).  The `-balanced` option is another discussion; I'm *strongly* against having that one available, we originally *took it out* for good reasons as well.  I'll argue that later as a separate comment on that change, so things don't get too messy here.

....but!  That would then only leave 2 default options, and 2 "advanced" ones, which is maybe a bit silly to have separate categories for...?  If so indeed, and if still putting them together in a single one, I'd argue for the default category, mostly because `-mask` is mandatory, but also because `-order` may in some scenarios be considered carefully, so shouldn't be "hidden" *too* far away. (as I at some point, I think, detailed when I made a very lengthy post about how `mtnormalise` works and why, `-order` can even be set to 0, which results in intensity normalisation without bias field correction: a very well defined mode of operation that can be useful in some pipelines, or even when using `mtnormalise` in different spots in existing pipelines)

So far my 2 (or more) cents on the categories.

Reply:
I'd have thought that if sub-dividing further, one might instead put `-mask` into its own option group, so that it appears at the top of the options list with its own bold banner highlighting that it's compulsory (which is rare for *MRtrix3* command-line options). But unless users have actually not understood that, it's probably not worth changing.